tempodb/encoding/common/bloom.go

```diff
@@ -2,41 +2,64 @@ package common
 
 import (
 	"bytes"
+	"math"
 
-	"github.com/grafana/tempo/pkg/util"
 	"github.com/willf/bloom"
+
+	"github.com/grafana/tempo/pkg/util"
 )
 
-const shardNum = 10
+const legacyShardCount = 10
 
 type ShardedBloomFilter struct {
 	blooms []*bloom.BloomFilter
 }
 
-func NewWithEstimates(n uint, fp float64) *ShardedBloomFilter {
-	b := &ShardedBloomFilter{
-		blooms: make([]*bloom.BloomFilter, shardNum),
+func evaluateK(shardSizeInBits, itemsPerBloom int) (k int) {
+	// Per https://llimllib.github.io/bloomfilter-tutorial/ under "How many hash functions should I use?"
+	// the optimal value of k: (m/n)ln(2)
+	// m: number of bits in the filter
+	// n: estimated number of objects
+	// k: number of hash functions
+	k = int(math.Ceil((float64(shardSizeInBits) / float64(itemsPerBloom)) * (math.Ln2)))
+
+	return
+}
+
+// NewBloom creates a ShardedBloomFilter
+func NewBloom(fp float64, shardSize, estimatedObjects uint) *ShardedBloomFilter {
+	// estimate the number of shards needed. an approximate value is enough
+	var shardCount uint
+	var kPerBloom uint
+	for {
+		shardCount++
+		b := bloom.New(shardSize*8, uint(evaluateK(int(shardSize*8), int(estimatedObjects/shardCount))))
```

Comment:
Need to reconsider this. I could alternately write some math formulae that estimate it, its just easier this way but tests by creating new blooms

Reply:
Swapped this out. Using the package function `bloom.EstimateParameters` instead.